app/src/lib/resolve-git-proxy.ts

```diff
@@ -0,0 +1,49 @@
+import { remote } from 'electron'
+import { enableAutomaticGitProxyConfiguration } from './feature-flag'
+import { parsePACString } from './parse-pac-string'
+
+export async function resolveGitProxy(
+  url: string
+): Promise<string | undefined> {
+  if (!enableAutomaticGitProxyConfiguration()) {
+    return undefined
+  }
+
+  // resolveProxy doesn't throw an error (at least not in the
+  // current Electron version) but it could in the future and
+  // it's also possible that the IPC layer could throw an
+  // error (if the URL we're given is null or undefined despite
+  // our best type efforts for example).
+  // Better safe than sorry.
+  const pacString = await remote.session.defaultSession
+    .resolveProxy(url)
+    .catch(err => {
+      log.error(`Failed resolving proxy for '${url}'`, err)
+      return 'DIRECT'
```

Comment:
`DIRECT` is PAC-speak for "there's no proxy, go on and contact the service directly". The `resolveProxy` method can't return undefined and if I've understood the chromium code it can't throw an error either meaning that if it fails to resolve a proxy it'll default to no-proxy (i.e. 'DIRECT'). 

The catch here is just an extra safe-guard for the future in case that method starts throwing all of a sudden (more likely that the IPC layer would though).

Reply:
gotcha, thanks for clarifying! that makes sense